build-system/tasks/helpers.js

```diff
@@ -817,6 +850,42 @@ async function getDependencies(entryPoint, options) {
   return Object.keys(result.metafile?.inputs ?? {});
 }
 
+/**
+ * @param {*} sourcemapsFile
+ * @param {*} options
+ * @return {*}
+ */
+function massageSourcemaps(sourcemapsFile, options) {
+  const sourcemaps = JSON.parse(sourcemapsFile);
+  sourcemaps.sources = sourcemaps.sources.map((source) => {
+    if (!source.startsWith('../')) {
+      return source;
+    }
+    return source.slice('../'.length);
+  });
+  if (sourcemaps.file) {
+    sourcemaps.file = path.basename(sourcemaps.file);
+  }
+
+  const extra = {
+    sourceRoot: getSourceMapBase(options),
+    includeContent: !!argv.full_sourcemaps,
+  };
+  return JSON.stringify({...sourcemaps, ...extra});
+}
+
+/**
+ * Returns whether or not we should compile with Closure Compiler.
+ * @return {boolean}
+ */
+function shouldUseClosure() {
+  // Normally setting this server-side experiment flag would be handled by
+  // the release process automatically. Since this experiment is actually on the build system
+  // itself instead of runtime, it is never run through babel (where the replacements usually happen).
+  // Therefore we must compute this one by hand.
+  return argv.define_experiment_constant !== 'EsbuildCompilation';
+}
```

Comment:
Maybe this should live in `build-system/compile/helpers.js`, where all the other closure functionality lives?

Reply:
I think it makes more sense here, where the bundler selection logic lives